Add `branch` parameter to TA flaky-tests docs (#2888)
buildkite/docs: pages/apis/rest_api/analytics/_flaky_tests_query_strings.md

      <p class="Docs__api-param-eg"><em>Example:</em> <code>?search="User#find_email"</code>, <code>?search="/billing_spec"</code></p>
    </td>
  </tr>
  <tr>
    <th>
      <code>branch</code>
    </th>
    <td>

>       <span>Returns flaky tests for which a flake has been detected one or more times on <code>branch</code>.</span>

Reviewer: Slight rewording suggestion:

```suggestion
      <span>Returns flaky tests for flakes detected one or more times on the branch whose name is specified by the <code>branch</code> value.</span>
```
